### PR TITLE
KAFKA-9327: GroupMetadata metrics are not documented

### PR DESCRIPTION
**Pull request by dongjinleekr**
This commit adds documentation on following `GroupMetadata` gauges:

- `NumOffsets`
- `NumGroups`
- `NumGroupsPreparingRebalance`
- `NumGroupsCompletingRebalance`
- `NumGroupsEmpty`
- `NumGroupsStable`
- `NumGroupsDead`

cc/ @gwenshap

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dongjinleekr** [comment]
Retest this please.

File: docs/ops.html [hunk 1]
```diff
@@ -963,6 +963,21 @@ <h4><a id="remote_jmx" href="#remote_jmx">Security Considerations for Remote Mon
         <td>kafka.server:type=ReplicaManager,name=IsrExpandsPerSec</td>
         <td>See above</td>
       </tr>
+      <tr>
```
Comment (omkreddy) [inline docs/ops.html]: maybe we can add the new metrics at end of the table?

Comment (dongjinleekr) [reply docs/ops.html]: Well, I thought grouping `kafka.server:type=ReplicaManager`s would be better. Is there any policy?

Comment (omkreddy) [reply docs/ops.html]: there is no policy, but `ReplicaManager`, `GroupMetadataManager` are not related, right? they can be separate group.

**Comment by dongjinleekr** [comment]
@omkreddy Here it is: Rebased onto the latest trunk, and moved the new metrics into the end of table.

**Comment by dongjinleekr** [comment]
cc/ @hachikuji @mimaison @omkreddy

**Review by omkreddy — APPROVED** [review]
@dongjinleekr Thanks for the PR. LGTM.

**Pull request closed without merging**